public/app/core/components/RolePicker/RolePickerMenu.tsx

```diff
@@ -0,0 +1,585 @@
+        <CustomScrollbar autoHide={false} autoHeightMax="300px" hideHorizontalTrack>
+          <div className={customStyles.menuSection}>
+            <div className={customStyles.groupHeader}>Built-in roles</div>
+            <RadioButtonGroup
+              className={customStyles.builtInRoleSelector}
+              options={BuiltinRoleOption}
+              value={selectedBuiltInRole}
+              onChange={onSelectedBuiltinRoleChange}
+              fullWidth={true}
+            />
+          </div>
+          {showGroups && !!optionGroups.length ? (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {optionGroups.map((option, i) => (
+                  <RoleMenuGroupOption
+                    data={option}
+                    key={i}
+                    isSelected={groupSelected(option.value) || groupPartiallySelected(option.value)}
+                    partiallySelected={groupPartiallySelected(option.value)}
+                    onChange={onGroupChange}
+                    onClick={onMenuGroupClick}
+                  />
+                ))}
+              </div>
+            </div>
+          ) : (
+            <div className={customStyles.menuSection}>
+              <div className={customStyles.groupHeader}>Fixed roles</div>
+              <div className={styles.optionBody}>
+                {fixedRoles.map((option, i) => (
+                  <RoleMenuOption
+                    data={option}
+                    key={i}
+                    isSelected={
+                      !!(
+                        (option.uid && !!selectedOptions.find((opt) => opt.uid === option.uid)) ||
+                        !!selectedBuiltInRoles.find((role) => role.uid === option.uid)
+                      )
+                    }
+                    disabled={!!(option.uid && !!selectedBuiltInRoles.find((role) => role.uid === option.uid))}
+                    onChange={onChange}
+                    // onClick={onMenuGroupClick}
+                    hideDescription
+                  />
+                ))}
+              </div>
+            </div>
+          )}
+          {!!customRoles.length && (
+            <div>
+              <div className={customStyles.groupHeader}>Custom roles</div>
+              <div className={styles.optionBody}>
+                {customRoles.map((option, i) => (
+                  <RoleMenuOption
+                    data={option}
+                    key={i}
+                    isSelected={
+                      !!(
+                        (option.uid && !!selectedOptions.find((opt) => opt.uid === option.uid)) ||
+                        !!selectedBuiltInRoles.find((role) => role.uid === option.uid)
+                      )
+                    }
+                    disabled={!!(option.uid && !!selectedBuiltInRoles.find((role) => role.uid === option.uid))}
```

Comment:
```suggestion
                    disabled={!!(option.uid && selectedBuiltInRoles.some((role) => role.uid === option.uid))}
```
`some` will return a boolean, which fits this case better.